rust/chains/hyperlane-cosmos/src/libs/verify.rs

```diff
@@ -76,3 +77,33 @@ pub fn pub_to_addr(pub_key: Vec<u8>, prefix: &str) -> ChainResult<String> {
 
     Ok(addr)
 }
+
+/// encode H256 to bech32 address
+pub fn priv_to_binary_addr(priv_key: Vec<u8>) -> ChainResult<H160> {
+    let sha_hash = sha256_digest(SigningKey::from_slice(priv_key.as_slice())
+                                     .unwrap()
+                                     .public_key()
+                                     .to_bytes())?;
+    let rip_hash = ripemd160_digest(sha_hash)?;
+
+    Ok(H160::from_slice(rip_hash.as_slice()))
+}
```

Comment:
first of all, I couldnt find the something replacable components on `fuel` implementations. It might be on rebase results, however, merging those components makes code more complex imo